pkg/builder/local_build_executor.go

```diff
@@ -160,7 +161,12 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 			util.StatusWrap(err, "Failed to acquire build environment"))
 		return response
 	}
-	defer buildDirectory.Close()
+	defer func() {
+		err := buildDirectory.Close()
+		if err != nil {
+			log.Printf("Failed to clean up build environment: %v\n", err)
+		}
+	}()
```

Comment:
Can't we simply do:
```go
defer func() {
    if err := buildDirectory.Close(); err != nil {
        attachErrorToExecuteResponse(response, util.StatusWrap(err, "Failed to close build directory"))
    }
}()
```
?

Reply:
Yes. I have a commit to do just that, and am looking into a better way to avoid Bazel retrying and printing follow up errors to the user.

This is the commit message if you find it useful, or I can prune it if it is too much information.

```
    Send failure to clean up build directory to the client

    The client may choose to retry the action and then receive a different
    follow-up error.

        Action: http://.../remote-execution/blobs/sha256/action/6068c5370fdcca9dfb70936f5ea481cf53092c5c3ea328e351a69641d5c7fbfa-169/
        ExecuteResponse: {...,
            "status":{
                "code":13,
                "message":"Failed to close build directory: Failed to remove build directory \"6068c5370fdcca9d\": Failed to remove \"root/external/com_google_absl/absl/base/internal\": directory not empty"
            }}

        Action: http://.../remote-execution/blobs/sha256/action/6068c5370fdcca9dfb70936f5ea481cf53092c5c3ea328e351a69641d5c7fbfa-169/
        ExecuteResponse: {...,
            "status":{
                "code":13,
                "message":"Failed to acquire build environment: Failed to create build directory \"6068c5370fdcca9d\": Cannot create a file when that file already exists."
        }}
```

Reply:
Concurrency is also an important note here if we want to accurately describe the current state of affairs, or we can punt that and look at the required improvements instead.